File: socialview-core/src/com/hendraanggrian/socialview/SimpleURLSpan.java
package com.hendraanggrian.socialview;

import android.annotation.SuppressLint;
import android.os.Parcel;
import android.support.annotation.ColorInt;
import android.support.annotation.NonNull;
import android.text.Spannable;
import android.text.TextPaint;
import android.text.style.URLSpan;

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
@SuppressLint("ParcelCreator")
class SimpleURLSpan extends URLSpan {

    @NonNull private final Spannable spannable;
    @ColorInt private final int color;

    SimpleURLSpan(@NonNull Spannable spannable, @ColorInt int color) {
        super("");
        this.spannable = spannable;
        this.color = color;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(getURL());
    }

    @Override
    public String getURL() {
        return spannable.subSequence(spannable.getSpanStart(this), spannable.getSpanEnd(this)).toString();
    }

    @Override
    public void updateDrawState(TextPaint ds) {
        ds.linkColor = color;
        super.updateDrawState(ds);
    }
}

File: socialview-core/src/com/hendraanggrian/socialview/SimpleURLSpan.kt
package com.hendraanggrian.socialview

import android.annotation.SuppressLint
import android.os.Parcel
import android.support.annotation.ColorInt
import android.text.Spannable
import android.text.TextPaint
import android.text.style.URLSpan

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
@SuppressLint("ParcelCreator")
internal class SimpleURLSpan(private val spannable: Spannable, @ColorInt private val color: Int) : URLSpan("") {

    override fun writeToParcel(dest: Parcel, flags: Int) = dest.writeString(url)

    override fun getURL() = spannable.subSequence(spannable.getSpanStart(this), spannable.getSpanEnd(this)).toString()

    override fun updateDrawState(ds: TextPaint) {
        ds.linkColor = color
        super.updateDrawState(ds)
    }
}